src/xml/SVGFile.java

```diff
@@ -0,0 +1,155 @@
+package xml;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import model.Canvas;
+import model.Shape;
+import model.specific_path.*;
+import view.Presentation;
+import view.ShapeState;
+import xml.*;
+
+public class SVGFile implements Presentation{
+
+	private String fileName;
+	private String xmlcontent;
+    private ArrayList<Shape> shapes = new ArrayList<Shape>();
+    private Canvas canvas;
+
+	public static File file;
+	private static BufferedWriter out;
+	
+	/*
+	 * the path to the folder where svg files should be saved
+	 */
+	private static String filepath = "files" + File.separator + "%s.svg";
+
+	
+	public SVGFile(String fileName, String xmlcontent, Canvas canvas) {
+		this.fileName = fileName;
+		this.xmlcontent = xmlcontent;
+        this.canvas = canvas;
+	}
+
+    public SVGFile(String fileName, String xmlcontent) {
```

Comment:
Got it. BTW, this is cause by tab v.s. 4 spaces. Should all use spaces at the first time